src/lib.zig

```diff
@@ -3015,6 +3033,251 @@ pub const Lua = struct {
+            .Float, .ComptimeFloat => {
+                const casted: Number = @floatCast(value);
+                lua.pushNumber(casted);
+            },
+            .Pointer => |info| {
+                switch (info.size) {
+                    .One => {
+                        if (@typeInfo(info.child) == .Array) {
+                            if (@typeInfo(info.child).Array.child != u8) {
+                                @compileError("only u8 arrays can be pushed");
+                            }
+                            _ = lua.pushString(&(value.*));
+                        } else {
+                            if (info.is_const) {
+                                @compileLog(value);
+                                @compileError("Pointer must not be const");
+                            }
+                            lua.pushLightUserdata(@ptrCast(value));
+                        }
+                    },
+                    .C, .Many, .Slice => {
+                        if (info.child != u8) {
+                            @compileError("Only u8 slices (strings) are valid slice types");
+                        }
+                        if (info.sentinel) |sentinel| {
+                            const casted: *info.child = @ptrCast(@constCast(sentinel));
+                            if (casted.* != 0) {
+                                @compileError("Sentinel of slice must be a null terminator");
+                            }
+                            _ = lua.pushString(value);
+                        } else {
+                            var null_terminated = lua.allocator().dupeZ(u8, value) catch unreachable;
+                            defer lua.allocator().free(null_terminated);
+                            _ = &null_terminated; //required to override unused var error
+                            _ = lua.pushString(null_terminated);
+                        }
+                    },
+                }
+            },
+            .Bool => {
+                lua.pushBoolean(value);
+            },
+            .Optional, .Null => {
+                if (value == null) {
+                    lua.pushNil();
+                } else {
+                    lua.pushAny(value.?);
+                }
+            },
+            .Struct => |info| {
+                lua.createTable(0, 0);
+                inline for (info.fields) |field| {
+                    lua.pushAny(field.name);
+                    lua.pushAny(@field(value, field.name));
+                    lua.setTable(-3);
+                }
+            },
+            .Void => {},
+            else => {
+                @compileLog(value);
+                @compileError("Invalid type given");
+            },
+        }
+    }
+
+    /// Converts the specified index of the lua stack to the specified
+    /// type if possible and returns it
+    pub fn toAny(lua: *Lua, comptime T: type, raw_index: i32) !T {
+        const index = lua.absIndex(raw_index);
```

Comment:
Okay now I remember why I did this

It was because toStruct parsing would fail if it was given a relative index. Inside of toAny I call toStruct. If the user gives a relative index like `-1`. Then the toStruct function would try to parse whatever table was at index `-1`

However getting values from a table requires modifying the stack so the relative index was now wrong. That said, I think(?) this transformation is only really necessary inside of toStruct, not inside of toAny.

Reply:
Yes, I'm pretty sure the call to absIndex can be removed here